slack.go

```diff
@@ -0,0 +1,43 @@
+package slacts
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/nlopes/slack"
+)
+
+type SlackRequester interface {
+	SearchMessagesContext(context.Context, string, slack.SearchParameters) (*slack.SearchMessages, error)
+}
+
+type SlackClient struct {
+	Client SlackRequester
+}
+
+// NewSlackClient returns SlackClient object.
+// arg httpclient for using custom http.Client. So this can be nil.
+//
+// For example,
+// 	NewSlackClient('YOUR_TOKEN', urlfetch.Client(ctx))
+func NewSlackClient(token string, httpclient *http.Client) *SlackClient {
+	var opts []slack.Option
+
+	if httpclient != nil {
+		opts = []slack.Option{slack.OptionHTTPClient(httpclient)}
+	}
+
+	return &SlackClient{
+		Client: slack.New(token, opts...),
+	}
+}
+
+// CountQuery returns count of matches with query
+func (sc *SlackClient) CountQuery(ctx context.Context, q string) (int, error) {
```

Comment:
I do not want abbreviations for public method arguments.

Would you like to change the argument `q` to` query`?

Reply:
Thank you for your review!
I fixed it!

Reply:
Thanks.

I became very easy to understand! 😃